lib/analyzer/binary-version-extractors/openjdk-jre.ts

```diff
@@ -0,0 +1,48 @@
+import { Docker } from '../../docker';
+import { Binary } from '../types';
+
+export {
+  extract,
+  packageNames,
+};
+
+// todo: add common names
+const packageNames = ['java'];
+
+function extract(targetImage: string)
+  : Promise<Binary | null> | null {
+  return new Docker(targetImage).run('java', [ '-version' ])
+    .catch(stderr => {
+      if (typeof stderr === 'string' && stderr.indexOf('not found') >= 0) {
+        return '';
+      }
+      throw new Error(stderr);
+    })
+    .then(parseOpenJDKBinary);
+}
+
+function parseOpenJDKBinary(fullVersionOutput: string) {
+  /*
+  `java -version` output:
+  `java version "1.8.0_191"
+  Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
+  Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)`
+  extracting `1.8.0_191-b12`
+  */
+  if (!fullVersionOutput) {
+    return null;
+  }
+  const bracketsRE = /\(build (.*)\)$/;
+  const jdkVersionLines = fullVersionOutput.split('\n');
+  const buildVersion = jdkVersionLines && jdkVersionLines.length > 1 &&
```

Comment:
I might be even stricter and say:
 if (jdkVersionLines.length) !== 3 {
   return null;
}

Reply:
for some reason the length is 4 but I did adjust it, let me know what you think